simple_api/object/validators.py

```diff
@@ -0,0 +1,71 @@
+from inspect import isclass
+
+
+def build_validation_fn(action_validators, parameter_validators, data_validators):
+    """
+    Prepares function for validation of input values and the action overall, instantiates validators which are not
+    already instantiated. Action_validators can be a sigle validator, Field_validators a List of tuples
+    (string, (validators...)) as prepared in object/actions.py
+    """
+    if action_validators is None and parameter_validators is None and data_validators is None:
+        return None
+
+    if not isinstance(action_validators, (list, tuple)):
+        action_validators = action_validators,
+
+    ins_action_validators = []
+    ins_parameter_validators = {}
+    ins_data_validators = {}
+
+    # Validators instantiation if they are not already
+    for cls_or_inst in action_validators:
+        if isclass(cls_or_inst):
+            ins_action_validators.append(cls_or_inst())
+        else:
+            ins_action_validators.append(cls_or_inst)
+    for parameter_validators_key, parameter_validators_values in parameter_validators.items():
+        for validator in parameter_validators_values:
+            if isclass(validator):
+                ins_parameter_validators.setdefault(parameter_validators_key, []).append(validator())
+            else:
+                ins_parameter_validators.setdefault(parameter_validators_key, []).append(validator)
+    for data_validators_key, data_validators_values in data_validators.items():
+        for validator in data_validators_values:
+            if isclass(validator):
+                ins_data_validators.setdefault(data_validators_key, []).append(validator())
+            else:
+                ins_data_validators.setdefault(data_validators_key, []).append(validator)
```

Comment:
tohle je hrozny chunk kodu, nejspis to delaji dlouha jmena promennych...co znamena `parameter_validators_key`? `for param_name, validators in ...`  je podle me mnohem explicitnejsi, a navic konzistentni se zbytkem projektu

Reply:
Přejmenováno, kdyžtak řekni kdyby tam pořád něco nesedělo